docs/source/etcd-architecture.rst

```diff
@@ -0,0 +1,238 @@
+Calico etcd-based architecture
+==============================
+
+The current version of Calico is built around `etcd`_. etcd is a distributed,
+consistent key value store for shared configuration and service discovery with
+a focus on being simple, secure, fast, and reliable.
+
+In Calico, etcd is used as the data store and communication mechanism for all
+the Calico components. This data store contains all the information the various
+Calico components need to set up the Calico network.
+
+This document discusses the various pieces of the Calico etcd-based
```

Comment:
Given the positioning as "Calico Architecture", I think you can potentially remove the paragraph above this one and just start with "This document discusses..."